athena-snowflake/src/main/java/com/amazonaws/athena/connectors/snowflake/SnowflakeMetadataHandler.java

```diff
@@ -180,26 +194,29 @@ public SnowflakeMetadataHandler(DatabaseConnectionConfig databaseConnectionConfi
     @Override
     public Schema getPartitionSchema(final String catalogName)
     {
+        LOGGER.debug("getPartitionSchema: " + catalogName);
         SchemaBuilder schemaBuilder = SchemaBuilder.newBuilder()
                 .addField(BLOCK_PARTITION_COLUMN_NAME, Types.MinorType.VARCHAR.getType());
         return schemaBuilder.build();
     }
 
-    private Optional<String> getPrimaryKey(TableName tableName) throws Exception 
+    private Optional<String> getPrimaryKey(TableName tableName) throws Exception
     {
+        LOGGER.debug("getPrimaryKey tableName: " + tableName);
         List<String> primaryKeys = new ArrayList<String>();
         try (Connection connection = getJdbcConnectionFactory().getConnection(getCredentialProvider())) {
-            try (PreparedStatement preparedStatement = connection.prepareStatement(SHOW_PRIMARY_KEYS_QUERY + tableName.getTableName());
+            try (PreparedStatement preparedStatement = connection.prepareStatement(SHOW_PRIMARY_KEYS_QUERY + "\"" + tableName.getSchemaName() + "\".\"" + tableName.getTableName() + "\"");
```

Comment:
static escape char

Reply:
Given that we have so many places were we are escaping and doing string manipulation; I think its worth have sdk-level util class that handle all theses ad-hoc functions.

Reply:
added internal ticket.